### DISCUSSION TITLE
JSON List / Array Field

### DISCUSSION THREAD

**Discussion by Radiergummi** [laravel/nova-issues#6565] [Ideas & Feature Requests]
I often have a situation where I'd like to add a field for a list of values stored in a JSON array—for example, a list of scopes for an OAuth SSO integration, or a set of single-use tags, or similar.  
Right now, the only sensible ways to work with this are:
 - A mapping step: Have users edit the values in a text field with a separator character (e.g. a comma), then transform that before saving. Introduces the usual parsing problems, for example if a value contains a comma.
 - A JSON/code field: Allows to edit the JSON right away, but that feels scary to sales folks.
 - A repeater with a custom JSON preset and a single text field: I haven't actually tried that yet, but it feels an awful lot like over-engineering and too much UI for something that should be simple.

So instead, I would love to see a simple List/Array field that took arbitrary values and stored them as a JSON array; similar to the `KeyValue` field, just without the keys. Bonus points if values could be dragged-and-dropped to change their order, but I would be fine without.  
I imagine it to look something like this:
```php
\Laravel\Nova\Fields\List::make('Scopes', 'scope')
  ->min(1) // at least one list entry
  ->max(100) // no more than a hundred
;
```
That would accept and produce output like this:
```php
["openid", "profile", "email"]
```

Is this something anyone else has wished for, or found a better solution to than me?

**Comment by nw-b** [comment]
Would love a built-in field for this. 

Have been using https://github.com/dillingham/nova-items-field (and then, a Nova 4 fork https://github.com/blendbyte/nova-items-field, going to have to fork again for Nova 5) for editing arrays, but seems like a great candidate for 1st party support.
